doc/decap/subnet_decap_HLD.md

```diff
@@ -0,0 +1,194 @@
+  - [6 Module Design](#6-module-design)
+    - [6.1 Tunnel Specification](#61-tunnel-specification)
+    - [6.2 DB Schema](#62-db-schema)
+      - [6.2.1 CONFIG\_DB](#621-config_db)
+      - [6.2.2 APPL\_DB](#622-appl_db)
+      - [6.2.3 STATE\_DB](#623-state_db)
+    - [6.3 Orchestration Agent](#63-orchestration-agent)
+    - [6.4 VLAN Subnet Decap](#64-vlan-subnet-decap)
+      - [6.4.1 VLAN Subnet Decap Rule Generation](#641-vlan-subnet-decap-rule-generation)
+      - [6.4.2 Dual-ToR Considerations](#642-dual-tor-considerations)
+      - [6.4.3 Netscan VLAN Subnet Probing](#643-netscan-vlan-subnet-probing)
+    - [6.5 Subnet Decap Configuration Workflow](#65-subnet-decap-configuration-workflow)
+    - [6.6 CLI](#66-cli)
+  - [7 Warm Reboot Support](#7-warm-reboot-support)
+  - [8 Test Plan](#8-test-plan)
+
+## 1 Revision
+
+|  Rev  |    Date    |    Author     | Change Description |
+| :---: | :--------: | :-----------: | ------------------ |
+|  0.1  | 03/30/2024 | Longxiang Lyu | Initial version    |
+
+## 2 Scope
+
+This document describes the subnet decapsulation feature on T0 SONiC that allows Netscan to probe VLAN subnet IP addresses.
+
+## 3 Definitions/Abbreviations
+
+| Term    | Meaning                                       |
+| ------- | --------------------------------------------- |
+| VLAN    | virtual local area network                    |
+| DIP     | destination IP                                |
+| SIP     | source IP                                     |
+| Netscan | Azure service to detect network path failures |
+
+## 4 Overview
+
+In Azure, Netscan probes the network paths/devices by sending IPinIP traffic. The IPinIP packet crafted by the Netscan sender has the outer DIP equals the destination device Loopback address, and the inner DIP equals the IP address of the Netscan sender. When the IPinIP packet is routed to/received by the destination device, they will be decapsulated and the inner packet will be routed back to the Netscan sender.
+As of today, Netscan uses this IP-decap based probing to detect route blackholes in the Azure network. The limitation is that Netscan is only able to probe the networking switches without the capability to detect any route blackholes for host nodes, especially VLAN subnet IPs. Due to the fact that the host nodes don’t have native IP-decap functionality, it is more appropriate to implement the IP-decap functionality on T0 SONiC as SONiC supports IPinIP decapsulation, and T0 SONiC will respond to the Netscan probes on behalf of the host nodes to decapsulate the Netscan IPinIP probe packets with DIP as any VLAN subnet IPs.
+In this design, subnet decap is introduced to enhance SONiC with the capability to generate the decap rules for the VLAN subnet so IPinIP packets from Netscan with DIP as either VLAN subnet IPs could be decapsulated and forwarded back to the Netscan sender to allow Netscan to have the awareness of any possible route blackholes to those destinations.
```

Comment:
Could you please clarify, how the route blackholes are identified?

Reply:
Added.